src/chart/generateCategoricalChart.tsx

```diff
@@ -1499,28 +1473,41 @@ export const generateCategoricalChart = ({
      * @return {Null}                  null
      */
     handleMouseEnter = (e: React.MouseEvent) => {
-      const { onMouseEnter } = this.props;
       const mouse = this.getMouseInfo(e);
 
       if (mouse) {
         const nextState: CategoricalChartState = { ...mouse, isTooltipActive: true };
         this.setState(nextState);
         this.triggerSyncEvent(nextState);
 
+        const { onMouseEnter } = this.props;
         if (_.isFunction(onMouseEnter)) {
           onMouseEnter(nextState, e);
         }
       }
     };
 
-    triggeredAfterMouseMove = (e: React.MouseEvent): any => {
-      const { onMouseMove } = this.props;
+    triggeredAfterMouseMove = (e: MousePointer): any => {
       const mouse = this.getMouseInfo(e);
       const nextState: CategoricalChartState = mouse ? { ...mouse, isTooltipActive: true } : { isTooltipActive: false };
 
       this.setState(nextState);
       this.triggerSyncEvent(nextState);
 
+      const { onMouseMove } = this.props;
+      if (_.isFunction(onMouseMove)) {
+        onMouseMove(nextState, e);
+      }
+    };
+
+    accessibilityMouseHandler = (e: MousePointer) => {
+      const mouse = this.getMouseInfo(e);
+      const nextState: CategoricalChartState = mouse ? { ...mouse, isTooltipActive: true } : { isTooltipActive: false };
+      this.setState(nextState);
+      this.triggerSyncEvent(nextState);
+
+      // I don't understand why it is receiving keyboard or focus events and triggering mouse events
```

Comment:
@ckifer Can I know the reason?

Reply:
Not sure what you mean. This is for accessibility see https://master--63da8268a0da9970db6992aa.chromatic.com/?path=/docs/api-accessibility--docs

Click that chart then use the arrow keys to navigate the tooltip

Reply:
I was curious about why onMouseMove is triggered for virtual events, not real mouse events.(probably because there seems to be no way to trigger changes in tooltips...)  
Similarly, other events seem to function the same way... 
These unpredictable synthetic events with names similar to default handlers in recharts make it difficult to use.

Reply:
I'm not the original implementer of any of this so I have no idea. Totally agree that things are unpredictable and confusing, we just have to be very careful while untangling things.. Trying to prevent swaths of breaking changes to build more trust and confidence with consumers (previously things were randomly merged and released).

But yeah... I don't understand either 😅

Reply:
Can you remove this comment here for now? Else reword it to be more generic? (Don't use "I" since once this is merged no one will know who "you" are 😅)